Snippets/Snippets_6/Features/AdvancedSatelliteFeature.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace Snippets6.Features
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.Features;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Transports;
+
+    public class MyAdvancedSatelliteFeature : Feature
+    {
+        public MyAdvancedSatelliteFeature()
+        {
+            EnableByDefault();
+        }
+
+        #region AdvancedSatelliteFeatureSetup
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            PipelineSettings messageProcessorPipeline = context.AddSatellitePipeline("AdvancedSatellite", TransportTransactionMode.TransactionScope, PushRuntimeSettings.Default, "targetQueue");
+            
+            // register the critical error
+            messageProcessorPipeline.Register("AdvancedSatellite", b => new MyAdvancedSatelliteBehavior(b.Build<CriticalError>()),
```

Comment:
@mikeminutillo - it came from my sample. I was trying to show how to move over from what was originally an advanced Satellite.